vmaas_sync/repo_sync.go

```diff
@@ -7,18 +7,30 @@ import (
 
 func syncRepos(syncStart time.Time, modifiedSince *string) error {
 	// mark non-thirdparty repos known to vmaas
-	updateRepos, err := getUpdatedRepos(syncStart, modifiedSince, false)
+	redhatRepos, thirdpartyRepos, err := getUpdatedRepos(syncStart, modifiedSince)
 	if err != nil {
 		return err
 	}
 
-	if len(updateRepos) == 0 {
+	if len(redhatRepos) == 0 && len(thirdpartyRepos) == 0 {
 		return nil
 	}
 
-	err = database.Db.Exec("UPDATE repo SET third_party = false WHERE name in (?)", updateRepos).Error
+	err = database.Db.Exec("UPDATE repo SET third_party = false WHERE name in (?)", redhatRepos).Error
 	if err != nil {
-		return errors.WithMessage(err, "Updating repo third_party flag")
+		return errors.WithMessage(err, "Updating repo third_party flag for redhat contet")
 	}
+
+	err = database.Db.Exec("UPDATE repo SET third_party = true WHERE name in (?)", thirdpartyRepos).Error
```

Comment:
Can not you just set third_party=true all not Red Hat repos? You can remove one exec command then.

Reply:
Fixed